src/middleware/common.middleware.js

```
@@ -112,3 +113,203 @@
+ * @returns {void}
+ */
+export const createPaginationTemplateParams = (req, res, next) => {
+  const { resultsCount, urlSubPath, paginationPageLength } = req
+  if (typeof resultsCount !== 'number' || typeof paginationPageLength !== 'number' || !urlSubPath) {
+    logger.error('Missing or invalid pagination parameters', { resultsCount, urlSubPath, paginationPageLength })
+    return next(new Error('Invalid pagination parameters'))
+  }
+  let { pageNumber } = req.params
+  pageNumber = parseInt(pageNumber)
+
+  if (isNaN(pageNumber) || pageNumber <= 0) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)